pkg/ee/default-policy-controller/controller.go

```diff
@@ -0,0 +1,399 @@
+			util.SetClusterCondition(
+				cluster,
+				r.versions,
+				kubermaticv1.ClusterConditionDefaultPolicyBindingsControllerCreatedSuccessfully,
+				corev1.ConditionTrue,
+				"",
+				"",
+			)
+		}); err != nil {
+			return &reconcile.Result{}, err
+		}
+	}
+
+	return nil, kerrors.NewAggregate(errors)
+}
+
+func (r *Reconciler) ensurePolicyBinding(ctx context.Context, namespace string, template kubermaticv1.PolicyTemplate, cluster *kubermaticv1.Cluster) error {
+	// Check if the binding already exists
+	existingPolicyBindingList := &kubermaticv1.PolicyBindingList{}
+	if err := r.List(ctx, existingPolicyBindingList, ctrlruntimeclient.InNamespace(namespace)); err != nil {
+		return fmt.Errorf("failed to list policy bindings: %w", err)
+	}
+
+	// Check if we already have a binding for this template
+	var currentPolicyBinding *kubermaticv1.PolicyBinding
+	for _, binding := range existingPolicyBindingList.Items {
+		if binding.Spec.PolicyTemplateRef.Name == template.Name {
+			// Check if this is an enforced or default binding
+			policyEnforced, _ := strconv.ParseBool(binding.Annotations[kubermaticv1.AnnotationPolicyEnforced])
+			policyDefault, _ := strconv.ParseBool(binding.Annotations[kubermaticv1.AnnotationPolicyDefault])
+
+			if policyEnforced || policyDefault {
+				currentPolicyBinding = &binding
+				break
+			}
+		}
+	}
+
+	reconcilers := []reconciling.NamedPolicyBindingReconcilerFactory{
+		PolicyBindingReconciler(r.log, template, currentPolicyBinding, cluster),
+	}
+	return reconciling.ReconcilePolicyBindings(ctx, reconcilers, namespace, r.Client)
+}
+
+func PolicyBindingReconciler(logger *zap.SugaredLogger, template kubermaticv1.PolicyTemplate, existingBinding *kubermaticv1.PolicyBinding, cluster *kubermaticv1.Cluster) reconciling.NamedPolicyBindingReconcilerFactory {
+	return func() (string, reconciling.PolicyBindingReconciler) {
+		bindingName := template.Name
+		return bindingName, func(binding *kubermaticv1.PolicyBinding) (*kubermaticv1.PolicyBinding, error) {
+			delete(template.Annotations, corev1.LastAppliedConfigAnnotation)
+
+			annotations := template.Annotations
+			if annotations == nil {
+				annotations = make(map[string]string)
+			}
+
+			if template.Spec.Enforced {
+				annotations[kubermaticv1.AnnotationPolicyEnforced] = strconv.FormatBool(true)
+			}
+			if template.Spec.Default {
+				annotations[kubermaticv1.AnnotationPolicyDefault] = strconv.FormatBool(true)
+			}
+
+			// Copy labels from the template if any
+			if len(template.Labels) > 0 {
+				if binding.Labels == nil {
+					binding.Labels = make(map[string]string)
+				}
+				for k, v := range template.Labels {
+					binding.Labels[k] = v
+				}
+			}
+
+			binding.Annotations = annotations
+			binding.Spec = kubermaticv1.PolicyBindingSpec{
+				PolicyTemplateRef: corev1.ObjectReference{
+					Name: template.Name,
+				},
+			}
+
+			logger.Infof("Reconciling PolicyBinding %s/%s", binding.Namespace, binding.Name)
```

Comment:
No need for this, the reconciling framework will already emit a log message when it creates/update an object.